### PR TITLE
updates

### PR DESCRIPTION
**Pull request by zippy**
minor improvements to whitepaper, cleaning up TODO sections, etc.

### CODE REVIEW

File: holochain.tex [hunk 1]
```diff
@@ -154,19 +154,27 @@ \subsection{Data-Centric and Agent-Centric Systems}
...
+By contrast, for \sgit there is no such constraint on any $\chain_n$, $\chain_m$ in nodes $n$ and $m$ matching, as git's core intent is to allow different agents act autonomously and divergently on a shared code-base, which would be impossible if the states always had to match.   
 
-Our model of a distributed system makes one thing very clear about the difference between \sbtc and \sgit.  As the size of $\mathcal{X}_n$ grows, necessarily all nodes of \sbtc must grow in size, whereas this is not necessarily the case for \sgit.  Though this seems like a clear limitation, it comes as a direct consequence of the constraint of $\forall n,m \in N: \chain_n\eqbang\chain_m$. Interestingly this is actually considered Bitcoin's strength as it defines the heart of ``consensus" in the Blockchain world, i.e., a limitation and method to the achieve sameness of state of all nodes.
+Through the lens of the formalisim the some other aspects of \sgit can be understood as follows:
```
Comment (lucksus) [inline holochain.tex]: -the

File: holochain.tex [hunk 2]
```diff
@@ -154,19 +154,27 @@ \subsection{Data-Centric and Agent-Centric Systems}
...
+\begin{enumerate}
+\item the validation function $V(e,v)$ by default only checks the structural validity of $e$ as a commit object not it's content (though note that git does also support signing of commits which is also part of the validation)
+\item the stimulus function $I(t)$ for \sgit consists of the set of git commands available to the user
```
Comment (lucksus) [inline holochain.tex]: missing space, should be:
```
\sgit\ consists
```

File: holochain.tex [hunk 3]
```diff
@@ -154,19 +154,27 @@ \subsection{Data-Centric and Agent-Centric Systems}
...
+\item the state transition function $\tau_\chain$ is the internal git function that adds a commit object and $\tau_\textrm{D}$ is the git function that adds code to the \texttt{index} triggered by \texttt{add}
+\item $E$ is, similarly to \sbtc, not formally defined for \sgit.
+\end{enumerate} 
+
+We leave a more in depth application of the formalism to \sgit as an excercise for the reader, however we underscore that the core difference between  \sbtc and \sgit lies in the formers constraint of $\forall n,m \in N: \chain_n\eqbang\chain_m$.  One direct consequence of this for \sbtc is that as the size of $\mathcal{X}_n$ grows, necessarily all nodes of \sbtc must grow in size, whereas this is not necessarily the case for \sgit, and in it lies the core of Bitcoin's scalability issues.
```
Comment (lucksus) [inline holochain.tex]: Several missing backslashes at the end of commands lead to missing spaces after math symbols.

File: holochain.tex [hunk 4]
```diff
@@ -154,19 +154,27 @@ \subsection{Data-Centric and Agent-Centric Systems}
...
+Perhaps a more accurate term for \sbtc's the hash-crack algorithm would be ``proof-of-luck" and for the process itself simply sameness, not consensus.  If you start from a data-centric viewpoint, which naturally throws out the ``experience" of all agents in favor of just one, it's much harder to design them to engage in processes that actually have the real-world properties of consensus. If the constraint of keeping all nodes' states the same were adopted consciously as a fit for a specific purpose, this would not be particularly problematic.  Unfortunately the legacy of this data-centric viewpoint has been held mostly unconsciously and is adopted by more generalized distributed computing systems, for which the intent doesn't specifically include the need to model ``digital matter" with universally absolute location.  While having the advantages of conceptual simplicity, it also immediately creates scalability issues, but worse, it makes it hard to take advantages inherent in the agent-centric approach.
```
Comment (lucksus) [inline holochain.tex]: -the

File: holochain.tex [hunk 5]
```diff
@@ -179,9 +187,12 @@ \section{Generalized Distributed Computation}
...
+Ethereum\footnote{https://github.com/ethereum/wiki/wiki/White-Paper} provides the current premier example of generalized distributed computing using the Blockchain model. The Ethereum approach comes from an ontology of replicating the data certainty of single physical computer, on top of the stratum of a bunch distributed nodes using the blockchain strategy of creating a single data reality in a cryptographic chain, but commiting computations, instead of just monetary transactions as in bitcoin, into the blocks.
```
Comment (lucksus) [inline holochain.tex]: ...a bunch +of distrbibuted nodes...

File: holochain.tex [hunk 6]
```diff
@@ -242,25 +249,25 @@ \subsection{Holochain}
...
 
-\item Let $\nu(A_n,q)=\eta_n$ be the set of $q$ nodes $\{n_1,n_2,\dots,n_q\}$ closest to $n$, so that
-$\forall n_i \in \eta_n, n_0 \notin \eta_n:$
+\item Let $r$ be a parameter of $\hcdht$ to be set dependent on the characteristics deemed beneficial for maintaining multiple copies of entries in the $DHT$ for the given application.
+Call $r$ the \term{resilience factor}.
+
+\label{dht:metrics}
+\item Allow that each node can mantain a set $M = \{m_n, \dots \}$ of metrics $m_n$ about other nodes, where each $m_n$ contains both a node's direct experience of $n$ with respect to that metric, as well as the experience of other nodes' of $n$.  Enforce that one such metric kept is \textit{•}{uptime} which keeps track of the percentage of time time a node is experienced to be available.   Call the process of nodes sharing these metrics \term{gossip} and refer to \ref{sec:gossip} for details.
```
Comment (lucksus) [inline holochain.tex]: ...percentage of time -time a node is experienced...

File: holochain.tex [hunk 7]
```diff
@@ -242,25 +249,25 @@ \subsection{Holochain}
...
 
-\item Let $\nu(A_n,q)=\eta_n$ be the set of $q$ nodes $\{n_1,n_2,\dots,n_q\}$ closest to $n$, so that
-$\forall n_i \in \eta_n, n_0 \notin \eta_n:$
+\item Let $r$ be a parameter of $\hcdht$ to be set dependent on the characteristics deemed beneficial for maintaining multiple copies of entries in the $DHT$ for the given application.
+Call $r$ the \term{resilience factor}.
+
+\label{dht:metrics}
+\item Allow that each node can mantain a set $M = \{m_n, \dots \}$ of metrics $m_n$ about other nodes, where each $m_n$ contains both a node's direct experience of $n$ with respect to that metric, as well as the experience of other nodes' of $n$.  Enforce that one such metric kept is \textit{•}{uptime} which keeps track of the percentage of time time a node is experienced to be available.   Call the process of nodes sharing these metrics \term{gossip} and refer to \ref{sec:gossip} for details.   
+
+\item  Enforce that $\forall \delta \in \dhtstate$ each node maintains a set $V_/delta = \{n_1,\dots,n_q\}$ of nodes also close to each node expected to hold $\delta$ so that resiliency is maintained taking into account node uptimes according to:
```
Comment (lucksus) [inline holochain.tex]: $V_\delta$

File: holochain.tex [hunk 8]
```diff
@@ -242,25 +249,25 @@ \subsection{Holochain}
...
+Call the union of such sets $V_\delta$, from a given node's perspective, the \term{overlap list}.
 
-\item \label{hc:shards} Allow every node $n$ to discard every $\delta_x \in \dhtstate_n$ if the number of closer (with regards to $d(x,y)$) nodes
-\begin{equation}
-\rho_x=\left\vert\{n_i|d(A_{n_i},\delta_{x,key}) < d(A_n,\delta_{x,key}\}\right\vert
-\end{equation}
-is greater than the \term{redundancy factor} $r$.
-Note that this results in the network adapting to changes in topology and DHT state migrations by regulating the number of network-wide redundant copies of all $\delta_i\in\dhtstate$ to match $r$. This implies the existence of $r$ partitions of N called \term{shards}\footnote{This sharding inspired the Holochain name, because if you partition the network it still retains an image of the whole in the same way that cutting a hologram results in each part still containing the whole image.\cite{hologram}} that each (among all their nodes $N_{Shard_i} \subset N$) hold a complete redundant copy of $\dhtstate$ :
+\item \label{hc:shards} Allow every node $n$ to discard every $\delta_x \in \dhtstate_n$ if the number of closer (with regards to $d(x,y)$) nodes greater than $q$.
```
Comment (lucksus) [inline holochain.tex]: +is

File: holochain.tex [hunk 9]
```diff
@@ -452,10 +460,38 @@ \subsubsection{Membranes \& Provenance}
...
+consensus and therefore (computationally expensive) absolute truth because 1) any single node uses provenance to independently verify any single transaction with the sources involved in that transaction and 2) because each Holochain application runs independently of all others, they are inherently permissioned by application specific rules for joining and continuing participation in that application's network.
 These both provide the benefit that any given Holochain application can tune the expense of that validation to a contextually appropriate level.
 
 
+\subsubsection{Gossip \& World Model}
+\label{sec:gossip}
+
+So far, we have focused on those parts of the validation function $V$ used verify that elments of $\chain$.  However, maintaining system integrity in distributed systems also requires that nodes have mechanisms sharing information about nodes that have broken the validation rules so that they can be excluded from participation. Thare exist, additionally, forms of bad-acting that do not live in the content of a transaction but in the patterns of transacting that are detrimental to the system, for example, denial of service attacks.
```
Comment (lucksus) [inline holochain.tex]: used +to verify -that elemetns of $\chain$

File: holochain.tex [hunk 10]
```diff
@@ -452,10 +460,38 @@ \subsubsection{Membranes \& Provenance}
...
+
+
+More formally:
+
+\begin{enumerate}
+\item Recall that each node maintains a set $M$ of metrics $m$ about other nodes it knows about. Note that in terms of our formalism, this world model is part of each node's non-chain state data $D$.
+\item Let $m$ be a tuple of tuples: $((\mu,c)_\text{self},(\mu,c)_\text{others})$ which record an experience $\mu$ of a node with respect to a given metric and a confidence $c$ of that exprience, both as directly experienced or as "hearsay" recieved from other nodes.
+\item Allow a class of entries stored in $\chain_n$ be used also as a metric $m_w$ which act as a signed declaration of the experience of $n$ regarding some other node.  Call such entries \term{warrants}.  These warrants allow us to use the standard tooling of Holochain to make provenance based, verifyable claims about other nodes in the network, which propagate orthogonally from the usual DHT methods, via gossip to nodes that need "hear" about these claims so as to make decisions about interacting with nodes.
```
Comment (lucksus) [inline holochain.tex]: ...need +to ¨hear¨...

File: holochain.tex [hunk 11]
```diff
@@ -452,10 +460,38 @@ \subsubsection{Membranes \& Provenance}
...
+\item $\forall m \in M$ let the function $G_\text{with}(m)$ return a set of nodes important for a node to gossip \textbf{with} defined by a probabilistc weighting that information recieved from those nodes will result in changing $m_\text{other}$.
+\item $\forall m \in M$ let the function $G_\text{about}(m)$ return a set of nodes important for a node to gossip \textbf{about} defined by the properties of $m$.  
+\item Define subsets of $G_\text{with}(m)$ according a corelation with what it means to have low vs. high confidence value $c$:
```
Comment (lucksus) [inline holochain.tex]: ..accoring +to a coRrelation..

File: holochain.tex [hunk 12]
```diff
@@ -596,9 +632,9 @@ \section{Implementation}
 \item Lisp Virtual Machines: zygomys \\\url{https://github.com/glycerine/zygomys}.
 \end{enumerate}
 
-\todo{benchmarking tests}
+Additionally we have created a benchmarking suites to examine the processing, bandwidth and storage used in various scenarios, and compared these with Ethereum applications in similar scenarios.  These can be seen here: \\\url{https://github.com/holochain/benchmarks}
```
Comment (lucksus) [inline holochain.tex]: suiteS

**Comment by zippy** [comment]
sounds good.  I guess I was doing it against develop because I thought we would merge develop into whitepaper when we release, because the next release will actually implement the worldmodel stuff...  but we could publish earlier too.  So either way...

**Pull request merged**